docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,707 @@
+
+var dynamicLearningPipeline = new CategoricalEstimator(env, "DemographicCategory")
+    .Append(new ConcatEstimator(env, "Features", "DemographicCategory", "LastVisits"))
+    .Append(new FastTreeBinaryClassificationTrainer(env, "HasChurned", "Features", numTrees: 20));
+
+var dynamicModel = dynamicLearningPipeline.Fit(trainData);
+
+// Build the same learning pipeline, but statically typed.
+// First, transition to the statically-typed data view.
+var staticData = trainData.AssertStatic(env, c => (
+        HasChurned: c.Bool.Scalar,
+        DemographicCategory: c.Text.Scalar,
+        LastVisits: c.R4.Vector));
+
+// Build the pipeline, same as the one above.
+var staticLearningPipeline = staticData.MakeNewEstimator()
+    .Append(r => (
+        r.HasChurned,
+        Features: r.DemographicCategory.OneHotEncoding().ConcatWith(r.LastVisits)))
+    .Append(r => classification.Trainers.FastTree(r.HasChurned, r.Features, numTrees: 20));
+
+var staticModel = staticLearningPipeline.Fit(staticData);
+
+// Note that dynamicModel should be the same as staticModel.AsDynamic (give or take random variance from
+// the training procedure).
+```
+
+## I want to look at my model's coefficients
+
+Oftentimes, once a model is trained, we are also interested on 'what it has learned'. 
+
+For example, if the linear model assigned zero weight to a feature that we consider important, it could indicate some problem with modeling. The weights of the linear model can also be used as a poor man's estimation of 'feature importance'.
+
+In the static pipeline API, we provide a set of `onFit` delegates that allow introspection of the individual transformers as they are trained.
+
+This is how we can extract the learned parameters out of the model that we trained:
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// We know that this is a classification task, so we create a multiclass classification context: it will give us the algorithms
+// we need, as well as the evaluation procedure.
+var classification = new MulticlassClassificationContext(env);
+
+// Step one: read the data as an IDataView.
+// First, we define the reader: specify the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // The four features of the Iris dataset.
+        SepalLength: ctx.LoadFloat(0),
+        SepalWidth: ctx.LoadFloat(1),
+        PetalLength: ctx.LoadFloat(2),
+        PetalWidth: ctx.LoadFloat(3),
+        // Label: kind of iris.
+        Label: ctx.LoadText(4)
+    ),
+    // Default separator is tab, but the dataset has comma.
+    separator: ',');
+
+// Retrieve the training data.
+var trainData = reader.Read(new MultiFileSource(dataPath));
+
+// This is the predictor ('weights collection') that we will train.
+MulticlassLogisticRegressionPredictor predictor = null;
```

Comment:
Why do we use a `MulticlassLogisticRegressionPredictor` when the learner is SDCA? If all the linear learners map to the same predictor, should it be `MulticlassLinearPredictor`?

Reply:
This code actually works. The naming is indeed strange. Would you mind crfeating an issue on that?

---
In reply to: [220775781](https://github.com/dotnet/machinelearning/pull/960#discussion_r220775781) [](ancestors = 220775781)